### PR TITLE
[virtualize] Fix fast swipe backward

### PR DESCRIPTION
**Pull request by igorbt**
The problem when using virtualize HOC is when fast and continuously swiping back and reaching the edge of rendered slides. Then user may think that there are no more slides, because they are not rendered until user stops swiping and animation ends (well, actually after a 500ms timeout).

Rendering previous items right away when going backward and reaching the edge if fixing this issue.

I understand that this may kill the ongoing animation in this particular use-case, but I think it's more important to show user the content with some choppy animation, than to show a beautiful animation with no content.

To overcome this limitation of having content rendered "real-time" and have uninterrupted animations at the same time, I think slides should have absolute positioning. This way, new slides can be rendered and visually positioned at the left or right, without replacing existing DOM nodes that are currently animating.

### CODE REVIEW

**Comment by oliviertassinari** [comment]
@igorbt Thanks for this PR, but I don't think I'm gonna merge it until I figured out how to propely solve this issue. Well, at least without any tests.

File: src/virtualize.js [hunk 1]
```diff
@@ -132,7 +132,16 @@ export default function virtualize(MyComponent) {
       }
 
       this.setState(nextState, () => {
-        this.setWindowTimer();
+        // We are going backward, but reached first rendered item, let's render previous right away, with no timer
+        if (indexDiff < 0 && index > 0 && indexContainer === 0) {
```
Comment (oliviertassinari) [inline src/virtualize.js]: What's the goal of this check?
```js
 && index > 0
```

Comment (igorbt) [inline src/virtualize.js]: index > 0 checks if there are more slides to display to the left. Maybe that's unnecessary, but I'm double checking.

**Comment by igorbt** [comment]
@oliviertassinari, It would be great to keep smooth animations and fixing this issue., but I'm afraid that this would require quite a rework of the current solution. Do you think you have the time to do this?

File: src/virtualize.js (hunk 1 above)
Comment (oliviertassinari) [inline src/virtualize.js]: This issue can happen when the index is below zero. It would be better to remove it.

**Comment by oliviertassinari** [comment]
> quite a rework of the current solution

Much of the work should be on at the SwipeableViews level. We need a wait to add an offset to react-motion without killing the animation. It's basicaly introducting a discontinuity.
We should also make sure it's working on react native.

I dont' have much time now. Have you tried using the `overscanSlideCount` to reduce this issue?

Otherwise, if you add a unit tests for this behavior and a comment in the code regarding temporary
workaround. I will merge it.

**Comment by igorbt** [comment]
I tried `overscanSlideCount` before thinking of anything else, but unfortunately it didn't help for this specific use-case of continuously and fast swiping back the slides. The issue is happening after few more swipes, and that's not really helpful.

I will try to add an unit test for this behavior. Actually I tried to do this from the beginning, but didn't quite understood at first glance how the things works there. But I will try more.

**Comment by oliviertassinari** [comment]
Having a look at the `react-motion` documentation, I feel like the only options is to render twice. Once to apply the offset without transition and a second on to apply the transition.
That logic should start around [here](https://github.com/oliviertassinari/react-swipeable-views/blob/master/src/SwipeableViews.js#L251).

**Comment by oliviertassinari** [comment]
@igorbt Thanks for this PR, you raised some good points.
However, I don't think that the solution is good enough. I'm closing this PR.

**Comment by igorbt** [comment]
I agree, @oliviertassinari, that's not good solution for the problem, it's at most a workaround. I'm not using it anymore myself. Actually, I'm not using virtualize at all because of this bug. I hope you or someone else will have the time to work on the proper solution, maybe the one you mentioned above.

**Pull request closed without merging**